src/c/service.c

```diff
@@ -512,6 +520,26 @@ static void edgex_device_device_upload_obj (devsdk_service_t *svc, JSON_Object *
         iot_log_warn (svc->logger, "Device upload: Missing device profileName definition");
       }
     }
+    else if ((edgex_devmap_device_exists (svc->devices, dname) && svc->overwritedevices))
+    {
+      iot_log_info (svc->logger, "Device %s already exists, deleting and re-adding", dname);
+      edgex_metadata_client_delete_device_byname(svc->logger,&svc->config.endpoints,svc->secretstore,dname,err);
+      if (err->code!=0) {
+        iot_log_warn (svc->logger, "Device re-upload: failed to delete exising device");
+        return;
+      }
+      if (json_object_get_string (jobj, "profileName"))
```

Comment:
suggest we check that profileName is supplied as soon as we have the dname, ie before checking edgex_devmap_device_exists, to avoid code duplication